src/pages/patientView/clinicalInformation/ClinicalInformationMutationalSignatureTable.tsx

```diff
@@ -0,0 +1,119 @@
+
+interface IMutationalSignatureRow {
+    mutationalSignatureId:string;
+    [key:string]:string;
+};
+
+export default class ClinicalInformationMutationalSignatureTable extends React.Component<{}> {
+    public render() {
+        const sampleData: IMutationalSignature[] = fetchMutationalSignatureData();
+        const sampleInvertedData =  this.invertData(sampleData);
+        const tableData = this.prepareData(sampleData);
+        const columns:Column<IMutationalSignatureRow>[] = [{id: 'mutationalSignatureId'}, ...sampleInvertedData.columns].map((col) =>  (
+            {
+                name: col.id,
+                render: (data:IMutationalSignatureRow)=><span>{data[col.id]}</span>,
+                download: (data:IMutationalSignatureRow) => `${data[col.id]}`,
+                filter: (data:IMutationalSignatureRow, filterString:string, filterStringUpper:string) =>
+                    (data[col.id].toString().toUpperCase().indexOf(filterStringUpper) > -1)
+            }
+        ));
+        //columns[0].sortBy = (data)=>data.mutationalSignatureId;
+        return <MutationalSignatureTable
+            columns={columns}
+            data={tableData}
+            className={styles.sampleTable}
+            showPagination={false}
+            initialItemsPerPage={SHOW_ALL_PAGE_SIZE}
+            showColumnVisibility={false}
+            initialSortColumn="attribute"
+            initialSortDirection="asc"
+        />;
+    }
+
+    public invertData(sampleData: IMutationalSignature[]): IConvertedSamplesData{
+        interface IColumn {
+            id:string;
+        };
+
+        interface IAttrData {
+            [attrId:string]: {
+                [sampleId: string]: ClinicalAttribute | string;
+                clinicalAttribute: ClinicalAttribute;
+                id: string;
+            };
+        };
+
+        interface IConvertedSamplesData {
+            columns: IColumn[];
+            items: IAttrData;
+        }
+
+
+        let sampleInvertedDataByMutationalSignature = _(sampleData)
+            .groupBy(x => x.sampleId)
+            .map((value, key)  => ({id: key, clinicalData: value}))
+            .value();
+
+        let output: IConvertedSamplesData = {columns: [], items: {}};
+
+        sampleInvertedDataByMutationalSignature.forEach((sample) => {
+            const sampleId = sample.id;
+            output.columns.push({id: sampleId});
+            sample.clinicalData.forEach((clinicalData => {
+                output.items[clinicalData.mutationalSignatureId] = output.items[clinicalData.mutationalSignatureId] || {};
+                output.items[clinicalData.mutationalSignatureId][sampleId] = clinicalData.value;
+                output.items[clinicalData.mutationalSignatureId].id = clinicalData.mutationalSignatureId;
+            }))
+        })
+
+        return output;
+    }
+
+    public prepareData(sampleData: IMutationalSignature[]) :IMutationalSignatureRow[] {
+
+        interface IMutationalSignatureRow {
```

Comment:
its a little messy to have named members and indexing, even though in this case it will be fine because no samples will have id `mutationalSignatureId`. I still think its cleaner to separate them in the data structure, eg `{ mutationalSignatureId:string, sampleValues:{[uniqueSampleKey:string]:number}`, 

we need `uniqueSampleKey` because `sampleId` may not be unique across studies, only `studyId`,`sampleId` pairs are unique, but pairs are hard to deal with, so we have `uniqueSampleKey` which is unique

also please pull this interface out of the function

Reply:
Agreed, I asked James follows the implementation from ClinicalInformationSamplesTable. We probably should do the same thing over there. @jxu8 could you fix ClinicalInformationSamplesTable as well?